ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -512,11 +511,15 @@ private[ember] object Parser {
     }
   }
 
-  private def combineArrays[A: scala.reflect.ClassTag](a1: Array[A], a2: Array[A]): Array[A] = {
-    val buff = mutable.ArrayBuffer[A]()
-    buff.++=(a1)
-    buff.++=(a2)
-    buff.toArray
+  private[this] def appendByteArrays(a1: Array[Byte], a2: Chunk[Byte]): Array[Byte] = {
+    val res = new Array[Byte](a1.length + a2.size)
+    var ix = 0
+    while (ix < a1.length) {
+      res(ix) = a1(ix)
+      ix += 1
+    }
```

Comment:
I feel stupid now, but we should use `System.arraycopy` for this one too 😆

Reply:
Well, isn't that one JVM-specific?

Reply:
It's not! :)

https://github.com/scala-js/scala-js/blob/33ed7588ad80eaff5802aaf3b6f620b408a200cd/javalanglib/src/main/scala/java/lang/System.scala#L97

https://github.com/scala-native/scala-native/blob/024a46917b5b5b835c200aa99227915b99fdbb34/javalib/src/main/scala/java/lang/System.scala#L23

Reply:
Oh, that's nice. Is it also "specialised", so it deals with arrays of bytes without boxing?

Reply:
I mean, it's a `System` intrinsic call. I think it just copies the underlying memory directly.

Reply:
Actually, personally I prefer the development style of adding commits and never rebasing or force-pushing. It's much easier to review and I find the history helpful when looking at old PRs.

But, I appreciate your contributions, so I won't complain :)